spec/helper.js

```diff
@@ -187,7 +187,7 @@ const reconfigureServer = async (changedConfiguration = {}) => {
   });
   cache.clear();
   parseServer = await ParseServer.startApp(newConfiguration);
-  Parse.CoreManager.setRESTController(RESTController);
+  // Parse.CoreManager.setRESTController(RESTController);
```

Comment:
Why has this been commented out?

Reply:
The issue occures when directAccess is set to true and the distinct query is wrapped within an eachBatch.

I wrote the test but for it to work I have to comment this line:

[spec/helper.js:190](https://github.com/parse-community/parse-server/blob/e944616ff7860ad4e86afa6dc39607a9599478e6/spec/helper.js#L190)

`
Parse.CoreManager.setRESTController(RESTController);
` 

The error occures because this is what naturally happens when parse server started with`directAccess` set to true:

[src/ParseServer:323](https://github.com/parse-community/parse-server/blob/e944616ff7860ad4e86afa6dc39607a9599478e6/src/ParseServer.js#L323)

`
 if (process.env.PARSE_SERVER_ENABLE_EXPERIMENTAL_DIRECT_ACCESS === '1' || directAccess) {
      Parse.CoreManager.setRESTController(ParseServerRESTController(appId, appRouter));
  }
 `
 
`ParseServer` rest controller is diffrent from `Parse-JS-SDK` I suppose which makes it access the AggregateController directly instead of SchemaController.

Reply:
Removed that comment and updated the test itself to use `ParseServerRESTController`.

Reply:
Please revert the change in the helper and instead call `restartServer` with the option `directAccess: true`.

Reply:
I did that already.